src/toil/batchSystems/slurm.py

```diff
@@ -68,9 +69,61 @@
     "STAGE_OUT",
     "STOPPED",
     "SUSPENDED"
-} 
+}
 
 class SlurmBatchSystem(AbstractGridEngineBatchSystem):
+    class PartitionInfo(NamedTuple):
+        partition_name: str
+        gres: bool
+        time: str
+        priority: str
+        cpus: str
+        memory: str
+
+    class PartitionSet:
+        """
+        Set of available partitions detected on the slurm batch system
+        """
+        default_gpu_partition: Optional[SlurmBatchSystem.PartitionInfo]
```

Comment:
Isn't `PartitionInfo` itself in scope here?

Reply:
I don't think so, both pycharm and mypy don't think it's within scope.